lib/components/widgets/wrapper/code_display.dart

```diff
@@ -0,0 +1,141 @@
+import 'package:dart_code_viewer_x/dart_code_viewer_x.dart';
+import 'package:flutter/material.dart';
+import 'package:fluttershow_base/components/helper/format_code.dart';
+import 'package:fluttershow_base/components/model/enum/code_display_themes.dart';
+
+/// A widget that displays formatted code
+/// with customizable styles and decorations.
+class CodeDisplay extends StatelessWidget {
+  /// Creates a [CodeDisplay] widget.
+  ///
+  /// The [code] parameter is the code to be displayed.
+  ///
+  /// The [codeColorTheme] parameter specifies
+  /// the color theme for syntax highlighting.
+  ///
+  /// The [width] and [height] parameters can be used
+  /// to set the dimensions of the widget.
+  ///
+  /// The [boxDecoration] parameter allows customizing
+  /// the box decoration of the code display.
+  ///
+  /// The [commentTextStyle], [baseTextStyle], [classTextStyle],
+  /// [keywordTextStyle], [stringTextStyle], and [punctuationTextStyle]
+  /// parameters can be used to customize
+  /// the styles of different parts of the code.
+  ///
+  /// The [padding] parameter specifies the padding of the code container.
+  CodeDisplay(
+    this.code, {
+    this.codeColorTheme,
+    this.width,
+    this.height,
+    this.boxDecoration,
+    this.commentTextStyle,
+    this.baseTextStyle,
+    this.classTextStyle,
+    this.keywordTextStyle,
+    this.stringTextStyle,
+    this.punctuationTextStyle,
+    this.padding,
+    super.key,
+  }) : formatedCode = formatCode(code);
+
+  /// The code to be displayed.
+  final String code;
+
+  /// The formatted code.
+  final String formatedCode;
```

Comment:
I am assuming this can be a private variable as it isn't being used anywhere outside. If this is used for testing, you could add an annotation `@visibleForTesting`.

Reply:
Changed to private 👍🏽